test/UnitTest/Microsoft.AspNet.OData.Test.Shared/Query/Expressions/SelectExpandWrapperJsonConverterTests.cs

```diff
@@ -0,0 +1,227 @@
+            selectExpandWrapper.ModelID = ModelContainer.GetModelID(model);
+
+            var selectExpandWrapperOfT = selectExpandWrapper as SelectExpandWrapper<SampleEntity>;
+            Assert.NotNull(selectExpandWrapperOfT);
+
+            selectExpandWrapperOfT.Instance = new SampleEntity
+            {
+                Property = "foobar"
+            };
+
+            var jsonSerializerOptions = new JsonSerializerOptions();
+            var converterFactory = new SelectExpandWrapperJsonConverter();
+            var converter = converterFactory.CreateConverter(typeof(T), jsonSerializerOptions) as JsonConverter<T>;
+
+            var memoryStream = new MemoryStream();
+            var utf8JsonWriter = new Utf8JsonWriter(memoryStream);
+
+            // Act
+            converter.Write(utf8JsonWriter, selectExpandWrapper, jsonSerializerOptions);
+            memoryStream.Position = 0;
+
+            var result = new StreamReader(memoryStream).ReadToEnd();
+
+            // Assert
+            Assert.Equal("{\"Property\":\"foobar\"}", result);
+        }
+
+        private static void DeserializePayloadUsingSelectExpandWriterJsonConverter<T>() where T : SelectExpandWrapper
+        {
+            // Arrange
+            var jsonSerializerOptions = new JsonSerializerOptions();
+            var converterFactory = new SelectExpandWrapperJsonConverter();
+            var converter = converterFactory.CreateConverter(typeof(T), jsonSerializerOptions) as JsonConverter<T>;
+
+            var utf8JsonReader = new Utf8JsonReader();
+
+            try
+            {
+                // Act
+                converter.Read(ref utf8JsonReader, typeof(object), jsonSerializerOptions);
+            }
+            catch (NotImplementedException ex)
+            {
+                // Assert
+                Assert.Equal($"'{typeof(T).Name}' is internal and should never be deserialized into.", ex.Message);
+                return;
+            }
+
+            // Assert - Fail the test if NotImplementedException was not thrown
+            Assert.True(false);
+        }
+
+        private static IEdmModel GetEdmModel()
+        {
+            ODataConventionModelBuilder builder = new ODataConventionModelBuilder();
+            builder.EntityType<SampleEntity>();
+            return builder.GetEdmModel();
+        }
+        
+        private class MockPropertyContainer : PropertyContainer
+        {
+            public MockPropertyContainer()
+            {
+                Properties = new Dictionary<string, object>();
+            }
+
+            public Dictionary<string, object> Properties { get; private set; }
+
+            public override void ToDictionaryCore(
+                Dictionary<string, object> dictionary,
+                IPropertyMapper propertyMapper,
+                bool includeAutoSelected)
+            {
+                foreach (var kvp in Properties)
+                {
+                    dictionary.Add(kvp.Key, kvp.Value);
```

Comment:
I thought the `propertyMapper` would be used to map the property names.

Reply:
@gathogojr This is a quick and dirty implementation of the `PropertyContainer`. The logic for the `PropertyContainer` and its derivations didn't change in this pull request.

Be that as it may, I should explain that for the tests in this file, it really doesn't matter.

If you look at the helper method named `SerializerPayloadUsingSelectExpandWriterJsonConverter`, we set the `UseInstanceForProperty` property of the `SelectExpandWrapper` to `true`
```csharp
selectExpandWrapper.UseInstanceForProperties = true;
```
By doing that, we're indicating that we want all the structural properties on the instance.
In addition, we also initialize the `Instance` property of the `SelectExpandWrapper<T>`:
```
selectExpandWrapperOfT.Instance = new SampleEntity
{
    Property = "foobar"
};
```
By doing that, the following logic in the `SelectExpandWrapper` will be executed. By design, it falls back on the instance:
- https://github.com/OData/WebApi/blob/b810e7185d45104f16fcd8ae1fc18cfeef923658/src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandWrapper.cs#L135
- https://github.com/OData/WebApi/blob/b810e7185d45104f16fcd8ae1fc18cfeef923658/src/Microsoft.AspNet.OData.Shared/Query/Expressions/SelectExpandWrapper.cs#L87

Reply:
@habbes I have gone further and modified the logic such that the `MockPropertyContainer` makes use of the `IPropertyMapper` to map property names. In addition, I added tests for advanced scenarios including those where `UseInstanceForProperties` is set to `false` and others where properties are decorated with `JsonProperty` attribute (Newtonsoft) and `JsonPropertyName` attribute (System.Text.Json) and both.